ProcessLib/LiquidFlow/LiquidFlowLocalAssembler-imp.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   LiquidFlowLocalAssembler.cpp
+ *
+ * Created on August 19, 2016, 2:28 PM
+ */
+
+#ifndef OGS_LIQUIDFLOWLOCALASSEMBLER_IMP_H
+#define OGS_LIQUIDFLOWLOCALASSEMBLER_IMP_H
+
+#include "LiquidFlowLocalAssembler.h"
+
+#include "NumLib/Function/Interpolation.h"
+#include "MaterialLib/PhysicalConstant.h"
+
+namespace ProcessLib
+{
+namespace LiquidFlow
+{
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+void LiquidFlowLocalAssembler<ShapeFunction, IntegrationMethod, GlobalDim>::
+    assemble(double const /*t*/, std::vector<double> const& local_x,
+             std::vector<double>& local_M_data,
+             std::vector<double>& local_K_data,
+             std::vector<double>& local_b_data)
+{
+    auto const local_matrix_size = local_x.size();
+    assert(local_matrix_size == ShapeFunction::NPOINTS);
+
+    auto local_M = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_M_data, local_matrix_size, local_matrix_size);
+    auto local_K = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_K_data, local_matrix_size, local_matrix_size);
+    auto local_b = MathLib::createZeroedVector<NodalVectorType>(
+        local_b_data, local_matrix_size);
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    const unsigned mat_id = 0;  // TODO for heterogeneous medium
+    const Eigen::MatrixXd& perm =
+        _material_properties.intrinsic_permeabiliy[mat_id];
+
+    // Note: For Inclined 1D in 2D/3D or 2D element in 3D, the first item in
+    //  the assert must be changed to perm.rows() == _element->getDimension()
```

Comment:
Do you mean Inclined 1D in 2D/3D or 2D element in 3D? I can do it, but it should be in done the general way in the integration of MathLib.